From f0d38dc51e53fde5dd1b2bad637eed7975c44df8 Mon Sep 17 00:00:00 2001
From: Calvin Johnson <calvin.johnson@nxp.com>
Date: Tue, 16 Jan 2018 11:17:46 +0530
Subject: [PATCH 089/129] armv8: fsl-lsch2: initialize pfe gemac

Call gemac_initialize to initialize both gemacs of pfe.

Signed-off-by: Calvin Johnson <calvin.johnson@nxp.com>
Signed-off-by: Anjaneyulu Jagarlmudi <anji.jagarlmudi@nxp.com>

Changes in v2: None
---
 arch/arm/cpu/armv8/fsl-layerscape/cpu.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/arch/arm/cpu/armv8/fsl-layerscape/cpu.c b/arch/arm/cpu/armv8/fsl-layerscape/cpu.c
index 15eefc7..ba1120d 100644
--- a/arch/arm/cpu/armv8/fsl-layerscape/cpu.c
+++ b/arch/arm/cpu/armv8/fsl-layerscape/cpu.c
@@ -21,6 +21,9 @@
 #include <efi_loader.h>
 #include <fm_eth.h>
 #include <fsl-mc/fsl_mc.h>
+#ifdef CONFIG_FSL_PFE
+#include <pfe_eth/pfe_eth.h>
+#endif
 #ifdef CONFIG_FSL_ESDHC
 #include <fsl_esdhc.h>
 #endif
@@ -478,6 +481,11 @@ int cpu_eth_init(bd_t *bis)
 {
 	int error = 0;
 
+#if defined(CONFIG_FSL_PFE)
+	gemac_initialize(bis, 0, "pfe_eth0");
+	gemac_initialize(bis, 1, "pfe_eth1");
+#endif
+
 #if defined(CONFIG_FSL_MC_ENET) && !defined(CONFIG_SPL_BUILD)
 	error = fsl_mc_ldpaa_init(bis);
 #endif
-- 
1.7.1

